docs/proposals/crt_management_proposal.md

```diff
@@ -0,0 +1,113 @@
+# Container Runtime Management
+
+## Description
+origin and openshift-ansible support multiple container runtimes.  This proposal
+is related to refactoring how we handle those runtimes in openshift-ansible.
+
+### Problems addressed
+We currently don't install docker during the install at a point early enough to
+not fail health checks, and we don't have a good story around when/how to do it.
+This is complicated by logic around containerized and non-containerized installs.
+
+A web of dependencies can cause changes to docker that are unintended and has
+resulted in a series of work-around such as 'skip_docker' boolean.
+
+We don't handle docker storage because it's BYO.  By moving docker to a prerequisite
+play, we can tackle storage up front and never have to touch it again.
+
+container_runtime logic is currently spread across 3 roles: docker, openshift_docker,
+and openshift_docker_facts.  The name 'docker' does not accurately portray what
+the role(s) do.
+
+## Rationale
+* Refactor docker (and related meta/fact roles) into 'container_runtime' role.
+* Strip all meta-depends on container runtime out of other roles and plays.
+* Create a 'prerequisites.yml' entry point that will setup various items
```

Comment:
By using the name prerequisites, will there be more things added to this than just the container runtime?

Reply:
Potentially.  I think this might be a good place to move other items, such as repos and subscriptions.  Those are typically a run-once type of thing, need to be in place before anything works, and probably don't need to run those tasks on every node on every play.

Reply:
I agree that pulling most run-once items to the front of runs is a more efficient pattern.  We just need to be careful that prerequisites don't become an entire separate huge install.  We should keep our eyes on the size and time of what prerequisites looks like but I'm in favor of this.